components/renku_data_services/data_connectors/orm.py

```diff
@@ -92,7 +92,7 @@ def dump(self) -> models.DataConnector:
         return models.DataConnector(
             id=self.id,
             name=self.name,
-            slug=self.slug.slug,
+            slug=self.slug.slug.lower(),
```

Comment:
Only a question: it looks like we need to take care to always lower things, when inserting and retrieving. Would it make sense to make sure to always lower on insert, so we can be sure to have valid data in the db? It would require less discipline. I was wondering if we could have a `Mapped[Slug]` instead of `Mapped[str]` for example (I saw we do this for `ULID`) and use its canonical string representation.

Reply:
The validation was added after some upper case slugs were already inserted. One solution could be to push a DB migration which would apply `lower()` on all existing slugs.